src/main/java/de/tum/in/www1/artemis/web/rest/TextAssessmentResource.java

```diff
@@ -0,0 +1,120 @@
+package de.tum.in.www1.artemis.web.rest;
+
+import com.codahale.metrics.annotation.Timed;
+import de.tum.in.www1.artemis.domain.*;
+import de.tum.in.www1.artemis.repository.TextSubmissionRepository;
+import de.tum.in.www1.artemis.service.*;
+import de.tum.in.www1.artemis.web.rest.util.HeaderUtil;
+import org.jetbrains.annotations.Nullable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * REST controller for managing TextAssessment.
+ */
+@RestController
+@RequestMapping("/api/text-assessments")
+public class TextAssessmentResource extends AssessmentResource {
+
+    private final Logger log = LoggerFactory.getLogger(TextAssessmentResource.class);
+
+    private static final String ENTITY_NAME = "textAssessment";
+
+    private final ParticipationService participationService;
+    private final ResultService resultService;
+    private final TextAssessmentService textAssessmentService;
+    private final TextExerciseService textExerciseService;
+    private final TextSubmissionRepository textSubmissionRepository;
+
+    public TextAssessmentResource(AuthorizationCheckService authCheckService,
+                                  ParticipationService participationService,
+                                  ResultService resultService,
+                                  TextAssessmentService textAssessmentService,
+                                  TextExerciseService textExerciseService,
+                                  TextSubmissionRepository textSubmissionRepository,
+                                  UserService userService) {
+        super(authCheckService, userService);
+
+        this.participationService = participationService;
+        this.resultService = resultService;
+        this.textAssessmentService = textAssessmentService;
+        this.textExerciseService = textExerciseService;
+        this.textSubmissionRepository = textSubmissionRepository;
+    }
+
+    @PutMapping("/exercise/{exerciseId}/result/{resultId}")
+    @PreAuthorize("hasAnyRole('TA', 'INSTRUCTOR', 'ADMIN')")
+    @Timed
+    public ResponseEntity<Result> saveTextAssessment(@PathVariable Long exerciseId, @PathVariable Long resultId, @RequestBody List<Feedback> textAssessments) {
+        TextExercise textExercise = textExerciseService.findOne(exerciseId);
+        ResponseEntity<Result> responseFailure = checkTextExerciseForRequest(textExercise);
+        if (responseFailure != null) return responseFailure;
+
+        Result result = textAssessmentService.saveAssessment(resultId, textAssessments);
+        return ResponseEntity.ok(result);
+    }
+
+    @PutMapping("/exercise/{exerciseId}/result/{resultId}/submit")
```

Comment:
If `/submit` is creating a new resource, shouldn't be a `POST` instead of a `PUT`?

Reply:
This is consistent with the Modelling Assessment.
The main reason why I think it should be `PUT` is that the request should be idempotent (e.g. with respect to overriding a submission)